product-service/src/handlers/getProduct.js

```diff
@@ -0,0 +1,15 @@
+import { getProductById } from '../services/getProductById/getProductById';
+import { buildHttpResponse } from '../utils/buildHttpResponse';
+
+export const getProduct = async (event, context) => {
+    const { id: productId = '' } = event.pathParameters;
+    try {
+      const product = await getProductById(productId);
+      if (!product) throw new Error('Product not found');
+      
+      return buildHttpResponse(200, product);
+    } catch (error) {
```

Comment:
What if we face any other error, except product not found. From my point of view, this case should be additionally protected with 500 error.